tests/unit/wc-payment-api/test-class-wc-payments-api-client.php

```diff
@@ -411,6 +411,74 @@ public function test_get_transaction_not_found() {
 		$this->payments_api_client->get_transaction( $transaction_id );
 	}
 
+	/**
+	 * Test a successful fetch of a single transaction.
+	 *
+	 * @throws Exception In case of test failure.
+	 */
+	public function test_list_transactions_success() {
+
+		$order_1 = WC_Helper_Order::create_order();
+		$order_1->update_meta_data( '_charge_id', 'ch_test_1' );
+		$order_1->save();
+
+		$order_2 = WC_Helper_Order::create_order();
+		$order_2->update_meta_data( '_charge_id', 'ch_test_2' );
+		$order_2->save();
+
+		$this->mock_db_wrapper
+			->expects( $this->any() )
+			->method( 'orders_with_charge_id_from_charge_ids' )
+			->with(
+				[
+					'ch_test_1',
+					'ch_test_2',
+				]
+			)
+			->will(
+				$this->returnValue(
+					[
+						[
+							'order'     => $order_1,
+							'charge_id' => 'ch_test_1',
+						],
+						[
+							'order'     => $order_2,
+							'charge_id' => 'ch_test_2',
+						],
+					]
+				)
+			);
+
+		$this->set_http_mock_response(
+			200,
+			[
+				'data' => [
+					[
+						'transaction_id' => 'txn_test_1',
+						'type'           => 'charge',
+						'charge_id'      => 'ch_test_1',
+					],
+					[
+						'transaction_id' => 'txn_test_2',
+						'type'           => 'charge',
+						'charge_id'      => 'ch_test_2',
+					],
+				],
+			]
+		);
+
+		$this->mock_no_subscriptions_for_order();
+
+		$transactions = $this->payments_api_client->list_transactions();
+
+		$this->assertEquals( 'txn_test_1', $transactions['data'][0]['transaction_id'] );
```

Comment:
We are typically using `assertSame` for better controlling types, please ensure the new tests are using this approach.